### ISSUE TITLE
FALSE POSITIVE: Active (40034 - .env Information Leak) alerting when not matching a valid .env file syntax

### ISSUE THREAD

**Issue by globeone** [zaproxy/zaproxy#6099]
**Describe the bug**
So, I looked around on the Internet for examples of .env files. They all have something in common. They contain an '=' equals symbol and are plain text `Content-Type: text/plain`. 
At the moment the .env Information Leak Active Scan rule seems to be only looking for an HTTP 200 response with content and not actually validating if it's come across a real .env file with these two attributes.

**To Reproduce**
1. Go to an API application that returns `HTTP 200 OK` `Content-Type: application/json`  `false` as to any invalid request. 
2. Click on Active Scan
3. Scroll down to Alerts
Active (40034 - .env Information Leak) detected

**Expected behavior**
If the = sign is missing and the content-type is not text/plain expecting the scanner to realize it's not a .env (Enviroment Variable File) and not throw an alert.

**Screenshots**
![2020-07-24_ envFalsePositiveImprovement](https://user-images.githubusercontent.com/10052587/88403905-6e418980-cdcd-11ea-814e-803f5286b3ff.png)

**Software versions**
 - ZAP: Weekly D-2020-7-20
 - Add-on: 
[2020-07-24ZAPWeeklyInstalledAddOns.xlsx](https://github.com/zaproxy/zaproxy/files/4972647/2020-07-24ZAPWeeklyInstalledAddOns.xlsx)

 - OS: Windows 10  [System.Environment]::OSVersion.VersionString
Microsoft Windows NT 10.0.18363.0
 - Java: Bellsoft Liberica JDK Full Including JavaFX
java -version
openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+13)
OpenJDK 64-Bit Server VM (build 14.0.2+13, mixed mode, sharing)
 - Browser: 
Version 84.0.4147.89 (Official Build) (64-bit)
78.0.2 (64-bits)

**Errors from the zap.log file**
Nothing related found in the log file. 

**Additional context**

From Docker
https://docs.docker.com/compose/env-file/

> Syntax rules
> These syntax rules apply to the .env file:
> 
> Compose expects each line in an env file to be in VAR=VAL format.
> Lines beginning with # are processed as comments and ignored.
> Blank lines are ignored.
> There is no special handling of quotation marks. This means that they are part of the VAL.

**Would you like to help fix this issue?**
Hopefully the suggestion to look for an "=" will help reduce this false positive.

**Comment by kingthorin** [comment]
Couple things:

1. This isn't limited to docker .env files, linux config files would be applicable as well.
2. I'm not sure that we can add a content type check, I'll have to put more thought into that. I'm worried that some sites/apps might HTML'ify (wrap) text files... Not counting it out, just want to think about it and discuss with others.
    - Your response claims to be json but is actually plain text.
3. I'm not sure `=` is applicable to all `potential .env` variants. As you quoted we should check for either `=` or `#` in the event that the particular assessed instance of the app is just a default with a lead-in comment or something (might be a bigger deal in other instances).

Anyway I think we can probably make some changes to improve things here, but more thought/discussion required.

**Comment by globeone** [comment]
Docker was not the only research I did, I checked NodeJS, Angular, and a few others Docker just had the nicest description of the syntax.

And you are correct about our devs cheating with application/json returning false as text/plain.
So maybe just checking for "="  or "#"  is enough to reduce the false positives.

Would it be possible to set this for High Threshold until the discussion is complete? Then I can take it for a spin as I test a lot of web apps with HTTP 200 OK False responses.
